x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -276,26 +287,40 @@ const TabContent = ({ posturetype }: { posturetype: PosturePolicyTemplate }) =>
   }
 
   if (shouldRenderNoFindings) {
-    return <NoFindingsStates postureType={posturetype} />;
+    return <NoFindingsStates postureType={selectedPostureTypeTab} />;
   }
 
   return (
     <CloudPosturePage query={getDashboardData}>
       <div data-test-subj={dataTestSubj}>
-        <IntegrationPostureDashboard
-          dashboardType={policyTemplate}
-          complianceData={getDashboardData.data}
-          notInstalledConfig={getNotInstalledConfig(policyTemplate, integrationLink)}
-          isIntegrationInstalled={setupStatus !== 'not-installed'}
-        />
+        <Routes>
+          <Route path={cloudPosturePages.cspm_dashboard.path}>
+            <IntegrationPostureDashboard
+              dashboardType={policyTemplate}
+              complianceData={getDashboardData.data}
+              notInstalledConfig={getNotInstalledConfig(policyTemplate, integrationLink)}
+              isIntegrationInstalled={setupStatus !== 'not-installed'}
+            />
+          </Route>
+
+          <Route path={cloudPosturePages.kspm_dashboard.path}>
+            <IntegrationPostureDashboard
+              dashboardType={policyTemplate}
+              complianceData={getDashboardData.data}
+              notInstalledConfig={getNotInstalledConfig(policyTemplate, integrationLink)}
+              isIntegrationInstalled={setupStatus !== 'not-installed'}
+            />
+          </Route>
+
+          {/* Redirect to default cspm dashboard if no match */}
+          {/* <Route path="*" render={() => <Redirect to={cloudPosturePages.cspm_dashboard.path} />} />*/}
```

Comment:
looks to be missing from code

Reply:
we should probably have a test case for this redirect i think

Reply:
Thanks! I will address the following code in follow-up PR.